### PR TITLE
DVCSMP-4714 weather icons

### PR DESCRIPTION
**Pull request by bflorian**
Fixed problem with leading zeros in icon strings. Also added refresh when app is updated (as in changing zip code)

### CODE REVIEW

**Comment by tpmanley** [comment]
I'm still getting a lux of 10000 for an `obs.iconCode` of 26

**Comment by tpmanley** [comment]
It's happening because `obs.iconCode` is an Integer. I added this code to the DTH:

`log.info "obs.iconCode is ${obs.iconCode.getClass()}"`

And got this: 

`java.lang.SecurityException: Invoking class java.lang.Integer.getClass() is not allowed @line 375 (estimateLux)`

**Comment by bflorian** [comment]
How about now @tpmanley ? Sandboxing prevents getting the Class object, which lets you do nasty things

File: devicetypes/smartthings/smartweather-station-tile.src/smartweather-station-tile.groovy [hunk 1]
```diff
@@ -347,11 +351,11 @@ private estimateLux(obs, sunriseDate, sunsetDate) {
...
     else {
         //day
-        switch(obs.iconCode) {
-            case '04':
+        switch(obs.iconCode as String) {
+            case '4':
                 lux = 200
                 break
-            case ['05', '06', '07', '08', '09', '10',
+            case ['5', '6', '7', '8', '9', '10',
```
Comment (tpmanley) [inline devicetypes/smartthings/smartweather-station-tile.src/smartweather-station-tile.groovy]: I still need to test this, but should 16 be in the list too? Also, if we changed the cases to integers we could do `case 5..26:` which seems like a nice improvement

Comment (bflorian) [reply devicetypes/smartthings/smartweather-station-tile.src/smartweather-station-tile.groovy]: 16 looks like snow so it seems like it should be included. I switched to using integer ranges.

**Review by tpmanley — APPROVED** [review]

**Comment by tpmanley** [comment]
@bflorian I think this can be merged

**Comment by tpmanley** [comment]
@bflorian @workingmonk any reason not to merge this, either now or after the next week's deploy (since this is pointed at `staging`)?

**Comment by rboy1** [comment]
This patch would work too, it also needs a fix to the 2 other attributes to convert to String while reporting them.

**Pull request merged**